app/models/manageiq/providers/lenovo/physical_infra_manager/operations.rb

```diff
@@ -54,9 +58,16 @@ def change_resource_state(verb, args, options = {})
                      :host => endpoint.hostname,
                      :port => endpoint.port)
 
-    # Turn on the location LED using the xclarity_client API
-    client.send(verb, args.ems_ref)
+    # Execute the action via the client
+    response = case verb
+               when :deploy_config_pattern
+                 client.send(verb, options[:id], [options[:uuid]], options[:restart], options[:etype])
+               else
+                 client.send(verb, args.ems_ref)
+               end
```

Comment:
Hm this feels wrong, what if we add another verb which doesn't match these arguments?
Maybe we should just yield the connection and let the caller define the arguments?

Reply:
@agrare Regarding how long the action takes, here we are just calling the client which sends a REST call to an LXCA instance to perform the action. The LXCA instance creates a task (in LXCA) and returns a REST response indicating if the request was successfully processed or not. Just as in the case of power operations (which can take awhile), we are not polling the task.

Reply:
Perfect @skovic that's exactly what I was wondering, since this is in a provision request we probably do want to poll on the task.

Reply:
So I think the undesireable thing here is the insertion of the case statement that does not have a good default. If someone calls another verb, we don't know if it should take the else clause.   As an alternative I recommend that we refactor the method appy_config_pattern method to make its own send call.  We can create a client_connection method that will create the connection and then  each of the public methods can define what method it wants to call on that connection directly.  I think change_resource_state does not add value now that we have broken the pattern of how the client is invoked.

Reply:
@agrare I updated the code based on @rodneyhbrown7 's suggestion. Will this work?